File: apps/web/src/lib/contribution-frequency.ts
export const CONTRIBUTION_FREQUENCY_LABELS: Record<string, string> = {
  DAILY: 'Daily',
  WEEKLY: 'Weekly',
  BI_WEEKLY: 'Bi-weekly',
  MONTHLY: 'Monthly',
};

export const CONTRIBUTION_FREQUENCY_OPTIONS = [
  { value: 'DAILY', label: 'Daily' },
  { value: 'WEEKLY', label: 'Weekly' },
  { value: 'BI_WEEKLY', label: 'Bi-weekly' },
  { value: 'MONTHLY', label: 'Monthly' },
] as const;

export function contributionFrequencyLabel(value?: string | null): string {
  if (!value) return '—';
  return CONTRIBUTION_FREQUENCY_LABELS[value] ?? value.replace(/_/g, ' ');
}

File: lib/core/constants/contribution_frequency.dart
/// Contribution schedule for Daily Savings and Child Savings accounts.
abstract final class ContributionFrequency {
  static const daily = 'DAILY';
  static const weekly = 'WEEKLY';
  static const biWeekly = 'BI_WEEKLY';
  static const monthly = 'MONTHLY';

  static const all = [daily, weekly, biWeekly, monthly];

  static String label(String value) {
    return switch (value) {
      daily => 'Daily',
      weekly => 'Weekly',
      biWeekly => 'Bi-weekly',
      monthly => 'Monthly',
      _ => value.replaceAll('_', ' '),
    };
  }

  static String description(String value) {
    return switch (value) {
      daily => 'Contribute every day',
      weekly => 'Once every week',
      biWeekly => 'Every two weeks',
      monthly => 'Once a month',
      _ => '',
    };
  }
}
